### PR TITLE
Wait until a newly created API key is ready to be used

### PR DESCRIPTION
**Pull request by gimenete**
Closes https://github.com/xataio/client-ts/issues/754

We do API requests to the data plane right after creating an API key, and that can be too fast and the API key might not be replicated yet.

With this PR we check up to 10 times and 10 seconds to see if the API is valid after that.

### CODE REVIEW

File: cli/src/commands/init/index.ts [hunk 1]
```diff
@@ -271,6 +274,27 @@ export default class Init extends BaseCommand {
...
+    const xata = await this.getXataClient();
+    const maxRetries = 10;
+    let retries = 0;
+    while (retries++ < maxRetries) {
+      try {
+        await xata.branches.getBranchList({ workspace, region, database });
+        return;
+      } catch (err) {
+        if (err instanceof Error && err.message.includes('Invalid API key')) {
+          this.info('Waiting until the new API key is ready to be used...');
```
Comment (exekias) [inline cli/src/commands/init/index.ts:288]: do we need to tell the user about this? normally this will take 1-2 seconds, maybe only telling them if retries > 2?

File: cli/src/commands/init/index.ts [hunk 2]
```diff
@@ -233,8 +233,11 @@ export default class Init extends BaseCommand {
 
     if (!apiKey) {
       apiKey = await createAPIKeyThroughWebUI();
+      this.apiKeyLocation = 'new';
```
Comment (exekias) [inline cli/src/commands/init/index.ts:236]: is this change intended? I don't see any other reference to `apiKeyLocation`

File: cli/src/commands/init/index.ts (hunk 1 above)
Comment (gimenete) [reply cli/src/commands/init/index.ts:288]: That's a good point. I agree with you but at the same time I don't want the UI to feel stuck if for some reason the replication takes longer.

File: cli/src/commands/init/index.ts (hunk 2 above)
Comment (gimenete) [reply cli/src/commands/init/index.ts:236]: It's intended. It is used to print this error message: https://github.com/xataio/client-ts/blob/697d75938be15e18d05c3a77f87d6ba364c5859b/cli/src/base.ts#L184-L189

For some reason it was missing.

File: cli/src/commands/init/index.ts (hunk 1 above)
Comment (gimenete) [reply cli/src/commands/init/index.ts]: What about this? https://github.com/xataio/client-ts/pull/769/commits/f816a77a449e7ccabab89dab11196db2da990b2e

Comment (exekias) [reply cli/src/commands/init/index.ts]: yeah, I don't know how easy would it be to keep adding dots or similar to make it explicit to the user that the CLI is alive, I'm trying to avoid user questions & uncertainties around this message, but maybe it's not a big deal.

anyway let's not block the PR on this

**Review by exekias — APPROVED** [review]

**Review by SferaDev — APPROVED** [review]

File: cli/src/commands/init/index.ts (hunk 1 above)
Comment (exekias) [reply cli/src/commands/init/index.ts]: oh, that works! 👍 thank you!

**Comment by gimenete** [comment]
@SferaDev that's great thinking. Let's merge this as a quick fix, and I'll think tomorrow about that. Thanks for the review!
